### PR TITLE
Update main.tf

### PR DESCRIPTION
**Pull request by tr0njavolta**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -2,12 +2,15 @@
 terraform {
   required_providers {
     azurerm = {
-      source  = "hashicorp/azurerm"
-      version = "~>1.31"
+      source = "hashicorp/azurerm"
```
Comment (im2nguyen) [inline main.tf:5]: Do we want to tag a version here? so the config works even when v3 is released?

File: main.tf [hunk 2]
```diff
@@ -29,7 +32,7 @@ resource "azurerm_subnet" "subnet" {
   name                 = "${var.prefix}TFSubnet"
   resource_group_name  = azurerm_resource_group.rg.name
   virtual_network_name = azurerm_virtual_network.vnet.name
-  address_prefixes     = ["10.0.1.0/24"]
+  address_prefix       = "10.0.1.0/24"
```
Comment (im2nguyen) [inline main.tf]: Is there a reason why you changed it to `address_prefix`? This argument is being deprecated for `address_prefixes`

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subnet#argument-reference

image

File: main.tf [hunk 3]
```diff
@@ -76,11 +79,6 @@ resource "azurerm_network_interface" "nic" {
   }
 }
 
-resource "azurerm_network_interface_security_group_association" "nic-nsg" {
```
Comment (im2nguyen) [inline main.tf:79]: Why did you remove this?

Comment (tr0njavolta) [reply main.tf:79]: I believe there was a dependency change in the az provider because it is not a requirement.

File: main.tf (hunk 2 above)
Comment (tr0njavolta) [reply main.tf]: I see that now, I'll revert

Comment (tr0njavolta) [inline main.tf]:
```suggestion
  address_prefixes       = ["10.0.1.0/24"]
```

File: main.tf (hunk 1 above)
Comment (nfagerlund) [reply main.tf:5]: Docs recommend that users always constrain provider versions, so yeah, we should probably try to set a good example. 😅

File: main.tf [hunk 4]
```diff
@@ -2,12 +2,16 @@
 terraform {
   required_providers {
     azurerm = {
-      source  = "hashicorp/azurerm"
-      version = "~>1.31"
+      source = "hashicorp/azurerm"
+      version = ">= 2.0"
```
Comment (tr0njavolta) [inline main.tf]:
```suggestion
      version = ">= 2.26"
```
